httputil/httputil.go

```diff
@@ -52,10 +79,13 @@ func NewSecureHttpClientWithCertificates(certificate string, key string) HttpCli
 
 	transport := &http.Transport{TLSClientConfig: tlsConfig}
 	httpClient := &http.Client{Transport: transport}
-	return &Client{httpClient}
+	return &Client{httpClient, retryBehavior}
```

Comment:
I think we should instead have a "RetryPolicy struct" which can be instanciated outside of the httpclient which will have the actual retry logic.